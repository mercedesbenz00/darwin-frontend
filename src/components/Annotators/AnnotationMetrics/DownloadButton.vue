<template>
  <button
    v-tooltip="downloadTooltip"
    type="button"
    class="download"
    :disabled="disabled"
    @click="onClick"
  >
    <download-icon />
  </button>
</template>

<script lang="ts">
import { Component, Prop, Vue } from 'vue-property-decorator'

import { DownloadIcon } from '@/assets/icons/V1'

@Component({ name: 'download-button', components: { DownloadIcon } })
export default class DownloadButton extends Vue {
  @Prop({ required: false, type: Boolean, default: false })
  disabled!: boolean

  get downloadTooltip () {
    if (!this.disabled) {
      return { content: 'Download' }
    }
    return { content: 'You don\'t have the permissions to download data' }
  }

  onClick () {
    this.$emit('click')
  }
}
</script>

<style lang="scss" scoped>
.download {
  border-radius: 50%;
  background: transparent;
  @include row;
  align-items: center;
  padding: 10px;

  transition: background-color .2s ease;

  &:disabled {
    opacity: .3;
  }
}

.download:not([disabled]):hover {
  background: $colorSecondaryLight2;
}

.download > * {
  width: 24px;
  height: 24px;
}
</style>
