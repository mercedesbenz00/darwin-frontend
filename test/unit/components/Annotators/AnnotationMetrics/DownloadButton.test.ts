import { shallowMount, createLocalVue } from '@vue/test-utils'

import DownloadButton from '@/components/Annotators/AnnotationMetrics/DownloadButton.vue'

const localVue = createLocalVue()
localVue.directive('tooltip', () => {})

let propsData: { disabled?: boolean }

describe('when enabled', () => {
  beforeEach(() => {
    propsData = {}
  })

  it('matches snapshot', () => {
    const wrapper = shallowMount(DownloadButton, { localVue, propsData })
    expect(wrapper).toMatchSnapshot()
  })

  it('emits click', () => {
    const wrapper = shallowMount(DownloadButton, { localVue, propsData })
    wrapper.find('button').trigger('click')
    expect(wrapper.emitted().click).toHaveLength(1)
  })
})

describe('when disabled', () => {
  beforeEach(() => {
    propsData = { disabled: true }
  })

  it('matches snapshot', () => {
    const wrapper = shallowMount(DownloadButton, { localVue, propsData })
    expect(wrapper).toMatchSnapshot()
  })

  it('never emits click', () => {
    const wrapper = shallowMount(DownloadButton, { localVue, propsData })
    wrapper.find('button').trigger('click')
    expect(wrapper.emitted().click).toBeUndefined()
  })
})
