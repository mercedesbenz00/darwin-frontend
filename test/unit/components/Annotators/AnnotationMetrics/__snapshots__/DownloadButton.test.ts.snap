// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when disabled matches snapshot 1`] = `
<button type="button" disabled="disabled" class="download">
  <download-icon-stub></download-icon-stub>
</button>
`;

exports[`when enabled matches snapshot 1`] = `
<button type="button" class="download">
  <download-icon-stub></download-icon-stub>
</button>
`;
